core/modules/installer/installer.browser.pages.inc

```diff
@@ -0,0 +1,998 @@
+  }
+
+  // Build the filters.
+  $backdrop_version = explode('.', BACKDROP_CORE_COMPATIBILITY);
+  $filters = array(
+    'version' => $backdrop_version[0],
+    'type' => $type,
+  );
+
+  // Add filters.
+  if (isset($_SESSION['installer_browser_text_filter_' . $type])) {
+    $filters['text'] = $_SESSION['installer_browser_text_filter_' . $type];
+  }
+  if (isset($_SESSION['installer_browser_order_filter_' . $type])) {
+    $filters['order'] = $_SESSION['installer_browser_order_filter_' . $type];
+  }
+  if (isset($_SESSION['installer_browser_sort_filter_' . $type])) {
+    $filters['sort'] = $_SESSION['installer_browser_sort_filter_' . $type];
+  }
+
+  $filters['items_per_page'] = 20;
+  $filters['page'] = isset($_GET['page']) ? $_GET['page'] : 0;
+
+  // Get the projects to display here based on the filters.
+  $results = installer_browser_fetch_results($filters);
+
+  // Save the listed projects in the session so it can be used.
+  $_SESSION['installer_browser_listed_projects'] = $results['projects'];
+
+  $list = array();
+  foreach ($results['projects'] as $project) {
+    $list[] = $project;
+  }
+
+  // Add the pager.
+  $total = $results['total'];
+  $items_per_page = 20;
+  $page = pager_default_initialize($total, $items_per_page);
+  $offset = $items_per_page * $page;
+  $start = ($total) ? $offset + 1 : 0;
+  $finish = $offset + $items_per_page;
+  if ($finish > $total) {
+    $finish = $total;
+  }
+
+  $sort_options = installer_browser_get_sort_options();
+  $current_order = isset($_SESSION['installer_browser_order_filter_' . $type]) ? $_SESSION['installer_browser_order_filter_' . $type] : 'score';
+  $current_sort = isset($_SESSION['installer_browser_sort_filter_' . $type]) ? $_SESSION['installer_browser_sort_filter_' . $type] : 'desc';;
+
+  $build = array();
+  $build['content'] = array(
+    'installer_browser_header' => array(
+      '#markup' => t('Showing @start to @finish of @total.', array(
+        '@start' => $start, '@finish' => $finish, '@total' => $total)),
+      '#weight' => 0,
+    ),
+    'installer_browser_sort_header' => array(
+      '#type' => 'item',
+      '#weight' => 2,
+      '#markup' => installer_browser_get_sort_widget($sort_options, $current_order, $current_sort),
+    ),
+    'installer_browser_list' => array(
+      '#markup' => theme('installer_browser_list', array('projects_list' => $list, 'type' => $type)),
+      '#weight' => 3,
+    ),
+    'pager' => array(
+      '#theme' => 'pager',
+      '#weight' => 99,
+    ),
+  );
+
+  return $build;
+}
+
+/**
+ * Page to display individual projects in a modal.
+ *
+ * @param $project
+ *   The project name.
+ *
+ * @return
+ *   Render array of a themed project.
+ */
+function installer_browser_project_display($project) {
+  $projects = $_SESSION['installer_browser_listed_projects'];
+  $form['project_dialog'] = array(
+    '#markup' => theme('installer_browser_project_dialog', array('project' => $projects[$project])),
+  );
+  return $form;
```

Comment:
Unless this is really necessary to use a renderable, seems like just returning the markup directly here would be a lot shorter.

```
return theme('installer_browser_project_dialog', array('project' => $projects[$project]));
```